pkg/controller/networkpolicy/clustergroup.go

```diff
@@ -249,6 +275,42 @@ func groupMembersComputedConditionEqual(conds []corev1a2.GroupCondition, conditi
 	return false
 }
 
+// processServiceReference processes the serviceRefernece in the group, and sets the group
+// selector based on the Service referenced.
+func (n *NetworkPolicyController) processServiceReference(group *antreatypes.Group) {
+	svcRef := group.ServiceReference
+	if svcRef == nil {
+		return
+	}
+	svc, err := n.serviceLister.Services(svcRef.Namespace).Get(svcRef.Name)
+	if err != nil || svc == nil {
+		if group.Selector == nil {
+			klog.V(2).Infof("Error getting Service object %s/%s, skip processing ClusterGroup %s", svcRef.Namespace, svcRef.Name, group.Name)
+		} else {
+			// Service was successfully synced for this Group but got deleted.
+			klog.V(2).Infof("Error getting Service object %s/%s, updating internal Group %s to empty selectors", svcRef.Namespace, svcRef.Name, group.Name)
+			group.Selector = nil
+		}
```

Comment:
it's probably just me but I don't understand the error processing. Shouldn't we just treat an error as the service not existing and unconditionally set `group.Selector` to `nil`? I am not sure why we have 2 cases.

Reply:
I put the two cases here because I wanted to have more specific debugging msgs for these two scenarios. But those definitely can be unified into one. Will update.